src/runtime_src/core/common/info_platform.cpp

```diff
@@ -156,8 +156,28 @@ add_controller_info(const xrt_core::device* device, ptree_type& pt)
                           % ((versionValue >> (1 * 8)) & 0xFF) // Minor
                           % ((versionValue >> (0 * 8)) & 0xFF)); // Version
     cmc.add("version", version);
-    cmc.add("serial_number", xrt_core::device_query<xq::xmc_serial_num>(device));
-    cmc.add("oem_id", xq::oem_id::parse(xrt_core::device_query<xq::oem_id>(device)));
+    std::string sn = xrt_core::device_query<xq::xmc_serial_num>(device);
+    if (sn.empty()) {
+      try {
+        sn = xrt_core::device_query<xq::hwmon_sdm_serial_num>(device);
+      }
+      catch (const xq::no_such_key&) {
+        // Ignoring if not available
+      }
+    }
+    cmc.add("serial_number", sn);
+
+    std::string oid = xq::oem_id::parse(xrt_core::device_query<xq::oem_id>(device));
+    if (oid.empty()) {
+      try {
+        oid = xrt_core::device_query<xq::hwmon_sdm_oem_id>(device);
```

Comment:
Hi @rozumx @AShivangi  hwmon_sdm driver also sends OEM_ID in Hex format like xmc's oem_id. 

So, shall I use existing xq::oem_id::parse() API to get the name for hwmon_sdm's oem_id or shall I introduce new API like xq::hwmon_sdm_oem_id::parse()?

I tried these two approaches and result is same - getting "Xilinx" as final output.

Reply:
I like using existing functionality whenever possible (don't like duplicate code, which this is starting to sound like).
So I vote for the first option.

Reply:
Done.